cypress/integration/manageLayers.cy.js

```diff
@@ -0,0 +1,198 @@
+        cy.waitForLayerContainers().then((elements) => {
+            cy.wrap(elements.length).should('equal', numberOfLayers + 1)
+        })
+
+        cy.log('verify the box of the layer is checked')
+        cy.getByDataTest('managelayers-button').click()
+        cy.waitForCheckbox(0, 'not.be.checked')
+        cy.getByDataTest('earthenginemodal-button').click()
+    })
+
+    it('without maps-management authority', () => {
+        cy.intercept(
+            'GET',
+            `${getApiBaseUrl()}/api/40/me?fields=id%2Cusername%2CdisplayName~rename(name)%2Cauthorities%2Csettings%5BkeyAnalysisDisplayProperty%5D`,
+            (request) => {
+                delete request.headers['if-none-match']
+                request.continue((response) => {
+                    if (response.body.authorities) {
+                        response.body.authorities = ['M_dhis-web-mapping']
+                    }
+                })
+            }
+        ).as('getAuthorization')
+
+        cy.wait('@getAuthorization', EXTENDED_TIMEOUT).then((interception) => {
+            cy.log(interception.response.body.authorities)
+            expect(interception.response.body.authorities).to.deep.equal([
+                //"F_PROGRAM_TRACKED_ENTITY_ATTRIBUTE_GROUP_ADD",
+                'M_dhis-web-mapping',
+            ])
+
+            Cypress.Commands.add('waitForLayerContainers', () => {
+                cy.getByDataTest('addlayerpopover', EXTENDED_TIMEOUT)
+                    .find('[class^="Layer_container"]')
+                    .should('have.length.greaterThan', numberOfLayers)
+            })
+            cy.getByDataTest('add-layer-button').click()
+            cy.waitForLayerContainers().then((elements) => {
+                cy.wrap(elements.length).should('equal', numberOfLayers + 1)
+            })
+            cy.getByDataTest('managelayers-button').should('not.exist')
+        })
+    })
+
+    it('datastore is not available', () => {
+        cy.request({
+            method: 'PUT',
+            url: `${getApiBaseUrl()}/api/40/dataStore/MAPS_APP/EARTH_ENGINE_LAYERS`,
+            headers: {
+                'Content-Type': 'application/json;charset=UTF-8',
+            },
+            body: {
+                not: 'expected',
+            },
+        }).then((response) => {
+            expect(response.status).to.eq(200)
+
+            cy.getByDataTest('add-layer-button').click()
+            cy.getByDataTest('addlayerpopover').should('be.visible')
+
+            cy.intercept(
+                'GET',
+                `${getApiBaseUrl()}/api/40/dataStore/MAPS_APP/EARTH_ENGINE_LAYERS`,
+                (request) => {
+                    delete request.headers['if-none-match']
+                }
+            ).as('getDataStore')
+
+            cy.wait('@getDataStore', EXTENDED_TIMEOUT).then((interception) => {
+                cy.log(interception.response)
+            })
+
+            cy.request({
```

Comment:
Could you add some comments to each block in this test to explain what's happening?

Reply:
chore: [da2968a](https://github.com/dhis2/maps-app/pull/3198/commits/da2968ad903b878ab66a55bdd6b8731bf6d69fc8)